cart/views.py

```diff
@@ -0,0 +1,99 @@
+from django.http import HttpResponse
+from django.shortcuts import render, redirect, reverse
+from django.contrib import messages
+
+from products.models import Product
+
+# Create your views here.
+
+def view_cart(request):
+    """A view that renders the cart contents page"""
+    return render(request, "cart/cart.html")
+    
+    
+def add_to_cart(request, item_id):
+    """Add a quantity of the specified product to the cart"""
+    
+    quantity = int(request.POST.get('quantity'))
+    size = None
+    if 'product_size' in request.POST:
+        size = request.POST['product_size']
+    cart = request.session.get('cart', {})
+    product = Product.objects.get(id=item_id)
+    redirect_url = request.POST.get('redirect_url')
+
+    # If item has sizes, organize by item_id and then size
+    if size:
+        if item_id in list(cart.keys()):
```

Comment:
What's the use of using `list()`?

Reply:
Also, to make the liens a bit shorter, I'd suggest preceding this with `item = cart.get(item_id)`, and then using `item` for this condition and all lines below

Reply:
Habit. Wrapping in list() returns a literal list, as opposed to just using dict.keys() which in python 3 returns an iterator (class dict_keys specifically) which doesn't support indexing. In our case it's actually not needed since we're not actually indexing anything there so we could use the iterator. I agree on the second point.